torch/fx/graph_module.py

```diff
@@ -82,6 +82,26 @@ def _copy_attr(from_module: torch.nn.Module, to_module: torch.nn.Module, target:
 
     setattr(to_module, field, getattr(from_module, field))
 
+# Assign attribute 'from_obj' to the qualified name 'target' on 'to_module
+# This installs empty Modules where none exist yet if they are subpaths of target
+def _assign_attr(from_obj: Any, to_module: torch.nn.Module, target: str):
+    *prefix, field = target.split('.')
+    for item in prefix:
+        t = getattr(to_module, item, None)
+        if from_obj is t:
```

Comment:
I don't think it can ever happen (because `from_obj` would be a thing at different level of hierarchy)

But it brings an important point: you should either ban nested names to appear in the dict at the same time (i.e. `foo` and `foo.bar`) or handle it properly in which case you need to enforce the ordering in which you do assignments below

Reply:
@dzhulgakov updated to sort from shortest to longest qualified names. I think this should do the right thing, let me know if there's something screwy